translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java

```diff
@@ -0,0 +1,363 @@
+    public void iGotKuraDataMessageWithPayloadBody(String payloadType) throws Throwable {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+
+        assertTrue(kuraDataMessage.getPayload().getBody().getClass().getSimpleName().equals(payloadType));
+    }
+
+    @And("^I got kura data message channel with \"(.+)\" and \"(.+)\" data$")
+    public void iGotKuraDataMessageChannelWithAndData(String scope, String clientId) {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+
+        assertTrue(kuraDataMessage.getChannel().getScope().equals(scope));
+        assertTrue(kuraDataMessage.getChannel().getClientId().equals(clientId));
+    }
+
+    @Then("^I got kura data message with proper payload metrics$")
+    public void iGotKuraDataMessageWithProperPayloadMetrics() {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+
+        assertTrue(kuraDataMessage.getPayload().getMetrics() != null);
+    }
+
+    @Then("^I got kura data message with empty payload$")
+    public void iGotKuraDataMessageWithEmptyPayload() {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+
+        assertEquals(null, kuraDataMessage.getPayload().getBody());
+    }
+
+    @Given("^I create jms message with (?:valid|invalid|empty) payload \"([^\"]*)\" and (?:valid|invalid) topic \"([^\"]*)\"$")
+    public void iCreateJmsMessageWithInvalidPayloadAndInvalidTopic(String payload, String topic) {
+        Date date = new Date();
+        JmsTopic jmsTopic = new JmsTopic(topic);
+
+        KuraPayload kuraPayload = new KuraPayload();
+        if (payload.equals("invalidPayload") || payload.equals("")) {
+            kuraPayload.setBody(payload.getBytes());
+        } else {
+            kuraPayload.getMetrics().put(payload, 200);
+        }
+
+        JmsPayload jmsPayload = new JmsPayload(kuraPayload.toByteArray());
+        JmsMessage jmsMessage = new JmsMessage(jmsTopic, date, jmsPayload);
+
+        stepData.put("JmsMessage", jmsMessage);
+    }
+
+    @And("^I try to translate jms message to kura data message$")
+    public void iTryToTranslateJmsMessageToKuraMessage() throws Exception {
+        JmsMessage jmsMessage = (JmsMessage) stepData.get("JmsMessage");
+
+        try {
+            KuraDataMessage kuraDataMessage = translatorDataJmsKura.translate(jmsMessage);
+            stepData.put("KuraDataMessage", kuraDataMessage);
+        } catch (KapuaException ex) {
+            verifyException(ex);
+        }
+    }
+
+    @And("^I try to translate kura data message to jms message$")
+    public void iTryToTranslateKuraDataMessageToJmsMessage() throws Exception {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+        JmsMessage jmsMessage = translatorDataKuraJms.translate(kuraDataMessage);
```

Comment:
insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.